packages/drivers/odsp-driver/src/test/fetchSnapshot.spec.ts

```diff
@@ -189,4 +201,213 @@ describe("Tests for snapshot fetch", () => {
+			requestUrl: siteUrl,
+		};
+		try {
+			await mockDownloadSnapshot(Promise.resolve(response), async () =>
+				service.getSnapshot({ groupIds: ["g1"] }),
+			);
+		} catch (error: any) {
+			assert.fail("the getSnapshot request should succeed");
+		}
+		assert(success, "should have asked for g1 group id");
+		assert(service["blobCache"].value.size === 0, "no blobs should be cached locally");
+		assert(service["commitCache"].size === 0, "no trees should be cached");
+	});
+
+	it("GetSnapshot() should not cache locally when specified in options", async () => {
+		async function mockDownloadSnapshot<T>(
+			_response: Promise<any>,
+			callback: () => Promise<T>,
+		): Promise<T> {
+			const getDownloadSnapshotStub = stub(fetchSnapshotImport, "downloadSnapshot");
+			getDownloadSnapshotStub.returns(_response);
+			try {
+				return await callback();
+			} finally {
+				getDownloadSnapshotStub.restore();
+			}
+		}
+		const snapshot: ISnapshot = {
+			blobContents,
+			snapshotTree: snapshotTreeWithGroupId,
+			ops: [],
+			latestSequenceNumber: 0,
+			sequenceNumber: 0,
+			snapshotFormatV: 1,
+		};
+		const odspResponse: IOdspResponse<Response> = {
+			content: (await createResponse(
+				{},
+				convertToCompactSnapshot(snapshot),
+				200,
+			)) as unknown as Response,
+			duration: 10,
+			headers: new Map([
+				["x-fluid-epoch", "epoch1"],
+				["content-type", "application/ms-fluid"],
+			]),
+			propsToLog: {},
+		};
+		const response: ISnapshotRequestAndResponseOptions = {
+			odspResponse,
+			requestHeaders: {},
+			requestUrl: siteUrl,
+		};
+		try {
+			await mockDownloadSnapshot(Promise.resolve(response), async () =>
+				service.getSnapshot({ groupIds: [], cacheSnapshot: false }),
+			);
+		} catch (error: any) {
+			assert.fail("the getSnapshot request should succeed");
+		}
+		const cachedValue = (await epochTracker.get(createCacheSnapshotKey(resolved))) as ISnapshot;
+		assert(cachedValue.snapshotTree.id === "SnapshotId", "snapshot should have been cached");
+		assert(service["blobCache"].value.size === 0, "no blobs should be cached locally");
+		assert(service["commitCache"].size === 0, "no trees should be cached");
+	});
 });
+
+const snapshotTreeWithGroupId: ISnapshotTree = {
+	id: "SnapshotId",
+	blobs: {},
+	trees: {
+		".protocol": {
+			blobs: {},
+			trees: {},
+		},
+		".app": {
+			blobs: { ".metadata": "bARD4RKvW4LL1KmaUKp6hUMSp" },
+			trees: {
+				".channels": {
+					blobs: {},
+					trees: {
+						default: {
+							blobs: {},
+							trees: {
+								dds: {
+									blobs: {},
+									trees: {},
+								},
+							},
+							groupId: "G3",
+						},
+					},
+					unreferenced: true,
+					groupId: "G2",
+				},
+				".blobs": { blobs: {}, trees: {} },
+			},
+			unreferenced: true,
+			groupId: "G4",
+		},
+	},
+};
+
+const blobContents = new Map<string, ArrayBuffer>([
+	[
+		"bARD4RKvW4LL1KmaUKp6hUMSp",
+		stringToBuffer(JSON.stringify({ summaryFormatVersion: 1, gcFeature: 0 }), "utf8"),
+	],
+]);
```

Comment:
As mentioned above, I think there is a bug in prefetchLatestSnapshot() of using []. It would be great to cover it and assert that we are asking for right thing on network

Reply:
Will add a test,